doc/src/build/programming-with-objects/ch5-child-objects.md

```diff
@@ -0,0 +1,248 @@
+public(script) fun remove_child(parent: &mut Parent, child: Child, ctx: &mut TxContext) {
+    let child_ref = Option::extract(&mut parent.child);
+    Transfer::transfer_child_to_address(child, child_ref, TxContext::sender(ctx));
+}
+```
+In the above function, the reference to the child is extracted from the `parent` object, which is then passed together with the `child` object to the `transfer_child_to_address`, with recipient as the sender of the transaction. It is important to note that this function must also take the `child` object as an argument. Move is not able to obtain the child object only from the reference. An object must always be explicitly provided in the transaction to make the transfer work. As we explaiend earlier, the fact that `transfer_child_to_address` requires the child reference as an argument helps guarantees that the `parent` object no longer holds a reference to the child object.
+
+#### transfer_child_to_object
+Another way to transfer a child object is to transfer it to another parent. The API is also defined in the Transfer library:
+```rust
+public fun transfer_child_to_object<T: key, R: key>(
+    child: T,
+    child_ref: ChildRef<T>,
+    owner: &mut R,
+): ChildRef<T>;
+```
+After this call, the object `child` will become a child object of the object `owner`.
+Comparing to the previous API, there are two primary differences:
+1. Instead of transferring the object to an recipient address, here we need to provide a mutable reference to the new parent objecet `owner`. Although we are not mutating the new parent objeect in practice, we require `mut` to make sure the new owner is not an immutable object: immutable objects cannot have child objects.
+2. The function returns a new `ChildRef`. This is because we are transferring this object to a new parent, and hence a new reference is created to represent this child relationship, which will be different from the old child reference.
+
+To see how to use this API, let's define a function that could transfer a child object to a new parent:
+```rust
+public(script) fun transfer_child(parent: &mut Parent, child: Child, new_parent: &mut Parent, _ctx: &mut TxContext) {
+    let child_ref = Option::extract(&mut parent.child);
+    let new_child_ref = Transfer::transfer_child_to_object(child, child_ref, new_parent);
+    Option::fill(&mut new_parent.child, new_child_ref);
+}
+```
+Similar to `remove_child`, the `child` object msut be passed explicitly by-value in the arguments. First of all we extract the existing child reference, and pass it to `transfer_child_to_object` along with `child`, and a mutable reference to `new_parent`. This call will return a new child reference. We then fill the `new_parent`'s `child` field with this new reference. Since `ChildRef` type is not droppable, `Option::fill` will fail if `new_parent.child` already contains an existing `ChildRef`. This ensures that we never accidentally drops a `ChildRef` without properly transferring the child.
```

Comment:
```suggestion
Similar to `remove_child`, the `child` object must be passed explicitly by-value in the arguments. First of all we extract the existing child reference, and pass it to `transfer_child_to_object` along with `child`, and a mutable reference to `new_parent`. This call will return a new child reference. We then fill the `new_parent`'s `child` field with this new reference. Since `ChildRef` type is not droppable, `Option::fill` will fail if `new_parent.child` already contains an existing `ChildRef`. This ensures that we never accidentally drop a `ChildRef` without properly transferring the child.
```